website/content/posts/evaluating-custom-predicates.md

```diff
@@ -0,0 +1,199 @@
+}
+
+_errors contains error if {
+    some attestation in _bacon_attestations
+    not attestation.statement.predicate.bacon_style
+    error := "Bacon attestation does not set the 'bacon_style' attribute"
+}
+
+_errors contains error if {
+    some attestation in _bacon_attestations
+    got := attestation.statement.predicate.bacon_style
+    got != _expected_style
+    error := sprintf("The bacon_style value %q is not the expected %q", [got, _expected_style])
+}
+
+_bacon_attestations := [attestation |
+    some attestation in input.attestations
+    attestation.statement.predicateType == "https://bacon/v42"
+]
+```
+
+(Check out the [docs](https://enterprisecontract.dev/docs/ec-policies/authoring.html) for more
+information on how to author policy rules 🔥)
+
+Next, we create a policy configuration, `policy.yaml`, to use the policy rules above.
+
+```yaml
+---
+sources:
+  - policy:
+      - /tmp/policies
+```
+
+Note: Change `/tmp/policies` to the absolute path to the directory where `bacon.rego` was saved.
+
+Finally, let's validate our image conforms to the policy config.
+
+```bash
+ec validate image --public-key cosign.pub --policy policy.yaml --image $REPOSITORY:latest \
+  --output yaml --show-successes
+```
+
+```yaml
+components:
+- attestations:
+  - predicateBuildType: https://localhost/dummy-type
+    predicateType: https://slsa.dev/provenance/v0.2
+    signatures:
+    - keyid: ""
+      sig: MEUCIFMtiIiz0h9+zJpc5MfwavZ2/BIxuhIig5uoePcQ+nOHAiEAzSCKAOH5irMG1bG5HNkVzZLOyDOV3SiIIrU6YCTz668=
+    type: https://in-toto.io/Statement/v0.1
+  - predicateType: https://bacon/v42
+    signatures:
+    - keyid: ""
+      sig: MEQCIHzmTK9YRU/PPfFjxRP6oSFNXyMIbAXEnQNP7GcCIjsbAiBVI8NtWYcvjg7/GmFC9Ce1e0XSh/mS5i5USHAX5I12tA==
+    type: https://in-toto.io/Statement/v0.1
+  containerImage: localhost:5000/ec-zero-to-hero@sha256:b5430d3d447434c795a508036e5046e41c009039be5b3f656f121c2426500d1e
+  name: Unnamed
+  signatures:
+  - keyid: ""
+    sig: MEQCIGIxpCboUYXF/fw6OuKmpM1Svi/0q+URD7oarLsji2+nAiBe3rgmWYOCa7sVpc2K5DKsef9hDigSlOHt6tl8v/8/JA==
+  source: {}
+  success: true
+  successes:
+  - metadata:
+      code: builtin.attestation.signature_check
+    msg: Pass
+  - metadata:
+      code: builtin.attestation.syntax_check
+    msg: Pass
+  - metadata:
+      code: builtin.image.signature_check
+    msg: Pass
+  - metadata:
+      code: bacon.style
+    msg: Pass
+ec-version: v0.3.2687-370a1a7
+effective-time: "2024-03-20T18:58:24.532182311Z"
+key: |
+  -----BEGIN PUBLIC KEY-----
+  MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEaxhfip26cuIfpDEVII0ilOHvFkee
+  igKbV5APr6OHYEstl9uJ8keON3VyzGfRB/2FUzr92J4ZN3YQUsTaiGc/HQ==
+  -----END PUBLIC KEY-----
+policy:
+  publicKey: cosign.pub
+  sources:
+  - policy:
+    - /tmp/policies
+success: true
+```
+
+Success!
+
+Change the value of the `_expected_style` variable in the `bacon.rego` file to any other value, e.g.
+`"chewy"`. Run the exact EC command again to see what a violation looks like.
```

Comment:
```suggestion
`"chunky"`. Run the exact EC command again to see what a violation looks like.
```

Chunky bacon is an obscure (old timer's) reference, see "_why the lucky stiff"

Reply:
https://poignant.guide/book/chapter-3.html

Reply:
(have to scroll down)

Reply:
Not a blocker obviously.. just a random suggestion to consider :grin:

Reply:
Well I obviously have to incorporate this now.